crates/Cargo.toml

```diff
@@ -0,0 +1,7 @@
+[workspace]
+
+members = [
+    "extracted_fzy",
+    "fuzzy_filter",
+    "icon",
+]
```

Comment:
I'd like to add Maple binary into that workspace too, but this could break some other code, and I wanted to cooperate with someone who knows what code could get broken (or at least test everything and find out if something gonna break).
@liuchengxu

Reply:
@ImmemorConsultrixContrarie 

> I'd like to add Maple binary into that workspace too.

I'm ok with that and I can test it.

To avoid the conflicts, I think it's better to start the new refactoring after #356 is merged, I have cherry-picked the code that are unrelated to the dynamic update in this PR into #356.